common/djangoapps/student/management/commands/transfer_students.py

```diff
@@ -52,7 +52,6 @@ class Command(TrackedCommand):
                     help="If True, try to transfer certificate items to the new course.")
     )
 
-    @transaction.commit_manually
```

Comment:
In this case, I think I lean towards 2) as well - all students transfer or no students transfer. However, that's different than the way the command currently works. And, unless it's an egregious error, we should likely stick with how it's always worked. So my final vote is for @symbolist 's change.